messagingpush/src/main/java/io/customer/messagingpush/util/PushTrackingUtil.kt

```diff
@@ -26,9 +26,16 @@ class PushTrackingUtilImpl(
         trackRepository.trackMetric(
             deliveryID = deliveryId,
             deviceToken = deliveryToken,
-            event = MetricEvent.opened
+            event = event
         )
-
         return true
     }
+
+    override fun parseLaunchedActivityForTracking(bundle: Bundle): Boolean {
+        return trackMetricEvent(bundle, MetricEvent.opened)
+    }
+
+    override fun parseIntentExtrasForTrackingDelivered(bundle: Bundle): Boolean {
+        return trackMetricEvent(bundle, MetricEvent.delivered)
+    }
```

Comment:
This looks like an overkill...

Reply:
Yes, `trackMetricEvent ` already takes in argument for `MetricEvent `, making 2 method that just gives different values doesn't provide much value for me

Reply:
Agreed, But `parseLaunchedActivityForTracking` is a public method and was the only reason for keeping it there. I can replace all calls with `trackMetricEvent`, but not going to delete `parseLaunchedActivityForTracking` in this PR. I'm going to mark it deprecated only, so we can remove it in future releases.

Reply:
I updated it to `parseAndTrackMetricEvent` for clarity. Also, I marked it as internal api because I think customer's probably don't really need to use this directly. But if you think differently, let me know so I can remove the annotation from it.